### PR TITLE
chore: port 5-app-infra readme to new format

### PR DESCRIPTION
**Pull request by mark1000**

### CODE REVIEW

File: 5-app-infra/README.md [hunk 1]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+1. Commit changes.
+   ```
+   git add .
+   git commit -m 'Your message'
+   ```
+1. Push your plan branch to trigger a plan for all environments.
+   ```
+   git push --set-upstream origin plan
+   ```
+ 1. Review the plan output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
```
Comment (daniel-cit) [inline 5-app-infra/README.md:146]:
```suggestion
 1. Review the plan output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md [hunk 2]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+1. Merge changes to development.
+   ```
+   git checkout -b development
+   git push origin development
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
```
Comment (daniel-cit) [inline 5-app-infra/README.md:152]:
```suggestion
1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md [hunk 3]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+1. Merge changes to non-production.
+   ```
+   git checkout -b non-production
+   git push origin non-production
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
```
Comment (daniel-cit) [inline 5-app-infra/README.md:158]:
```suggestion
1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md [hunk 4]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+1. Merge changes to production branch
+   ```
+   git checkout -b production
+   git push origin production
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
```
Comment (daniel-cit) [inline 5-app-infra/README.md:164]:
```suggestion
1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md [hunk 5]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
 
 ### Run terraform locally
 
 1. Change into 5-app-infra folder.
 1. Run `cp ../build/tf-wrapper.sh .`
 1. Run `chmod 755 ./tf-wrapper.sh`.
 1. Rename `common.auto.example.tfvars` to `common.auto.tfvars` and update the file with values from your environment and bootstrap.
```
Comment (daniel-cit) [inline 5-app-infra/README.md:171]:
```suggestion
1. Rename `common.auto.example.tfvars` to `common.auto.tfvars` and update the file with values from your environment and 0-bootstrap.
1. Rename `bu1-development.auto.example.tfvars` to `bu1-development.auto.tfvars` and update the file with values from your environment.
1. Rename `bu1-non-production.auto.example.tfvars` to `bu1-non-production.auto.tfvars` and update the file with values from your environment.
1. Rename `bu1-production.auto.example.tfvars` to `bu1-production.auto.tfvars` and update the file with values from your environment.
```

File: 5-app-infra/README.md [hunk 6]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+1. Copy terraform wrapper script to the root of your new repository.
+   ```
+   cp ../terraform-example-foundation/build/tf-wrapper.sh .
+   ```
+1. Ensure wrapper script can be executed.
+   ```
+   chmod 755 ./tf-wrapper.sh
+   ```
 1. Rename `common.auto.example.tfvars` to `common.auto.tfvars` and update the file with values from your environment.
```
Comment (daniel-cit) [inline 5-app-infra/README.md:133]:
```suggestion
1. Rename `common.auto.example.tfvars` to `common.auto.tfvars` and update the file with values from your environment and 0-bootstrap.
```

File: 5-app-infra/README.md [hunk 7]
```diff
@@ -1,12 +1,67 @@
...
+and permissions to start using the Cloud Foundation Toolkit (CFT). This
+step also configures a CI/CD pipeline for foundations code in subsequent
+stages.</td>
+</tr>
+<tr>
+<td><a
+href="https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/1-org">1-org</a></td>
+<td>Sets up top level shared folders, monitoring and networking projects, and
+organization-level logging, and sets baseline security settings through
+organizational policy.</td>
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:23]:
```suggestion
<td>Sets up top-level shared folders, monitoring and networking projects, 
organization-level logging, and baseline security settings through
organizational policies.</td>
```

File: 5-app-infra/README.md [hunk 8]
```diff
@@ -1,12 +1,67 @@
...
+href="https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/2-environments">2-environments</a></td>
+<td>Sets up development, non-production, and production environments within the
+Google Cloud organization that you've created.</td>
+</tr>
+<tr>
+<td><a
+href="https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/3-networks">3-networks</a></td>
+<td>Sets up base and restricted shared VPCs with default DNS, NAT (optional),
+Private Service networking, VPC service controls, on-premises Dedicated
+Interconnect, and baseline firewall rules for each environment. Also sets
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:36]:
```suggestion
Interconnect, and baseline firewall rules for each environment. It also sets
```

File: 5-app-infra/README.md [hunk 9]
```diff
@@ -1,12 +1,67 @@
...
+up the global DNS hub.</td>
+</tr>
+<tr>
+<td><a
+href="https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/4-projects">4-projects</a></td>
+<td>Set up a folder structure, projects, and application infrastructure pipeline for applications,
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:42]:
```suggestion
<td>Sets up a folder structure, projects, and an application infrastructure pipeline for applications,
```

File: 5-app-infra/README.md [hunk 10]
```diff
@@ -1,12 +1,67 @@
...
+</table>
+
+For an overview of the architecture and the parts, see the
+[terraform-example-foundation README](https://github.com/terraform-google-modules/terraform-example-foundation)
+file.
+
+## Purpose
+
+The purpose of this step is to deploy a simple [Compute Engine](https://cloud.google.com/compute/) instance in one of the business unit projects using the infra pipeline set up in 4-projects.
+The infra pipeline is created in step `4-projects` within the shared env and has a [Cloudbuild](https://cloud.google.com/build/docs) pipeline configured to manage infrastructure within projects. To enable deployment via this pipeline, the projects deployed should [enable](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/business_unit_1/development/example_base_shared_vpc_project.tf#L31-L32) `enable_cloudbuild_deploy` flag and provide the Cloud Build service account value via`cloudbuild_sa`.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:59]:
```suggestion
The infra pipeline is created in step `4-projects` within the shared env and has a [Cloud Build](https://cloud.google.com/build/docs) pipeline configured to manage infrastructure within projects. To enable deployment via this pipeline, the projects deployed should [enable](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/business_unit_1/development/example_base_shared_vpc_project.tf#L31-L32) the `enable_cloudbuild_deploy` flag and provide the Cloud Build service account value via `cloudbuild_sa`.
```

File: 5-app-infra/README.md [hunk 11]
```diff
@@ -1,12 +1,67 @@
...
+
+This enables the Cloud Build service account to impersonate the project service account and use it to deploy infrastructure. The roles required for project SA can also be [managed](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/business_unit_1/development/example_base_shared_vpc_project.tf#L30) via `sa_roles`. (Note: This requires per project SA impersonation. If you would like to have a single SA managing an environment and all associated projects, that is also possible by [granting](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/modules/single_project/main.tf#L62-L68) `roles/iam.serviceAccountTokenCreator` to an SA with the right roles in `4-projects/env`.)
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:61]:
```suggestion
This enables the Cloud Build service account to impersonate the project service account and use it to deploy infrastructure. The roles required for the project SA can also be [managed](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/business_unit_1/development/example_base_shared_vpc_project.tf#L30) via `sa_roles`. (Note: This requires per project SA impersonation. If you would like to have a single SA managing an environment and all associated projects, that is also possible by [granting](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/4-projects/modules/single_project/main.tf#L62-L68) `roles/iam.serviceAccountTokenCreator` to an SA with the right roles in `4-projects/env`.)
```

File: 5-app-infra/README.md [hunk 12]
```diff
@@ -1,12 +1,67 @@
...
 
 There is also a [Source Repository](https://cloud.google.com/source-repositories) configured with build triggers similar to [foundation pipeline](https://github.com/terraform-google-modules/terraform-example-foundation#0-bootstrap) setup in `0-bootstrap`.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:63]:
```suggestion
There is also a [Source Repository](https://cloud.google.com/source-repositories) configured with build triggers similar to the [foundation pipeline](https://github.com/terraform-google-modules/terraform-example-foundation#0-bootstrap) setup in `0-bootstrap`.
```

File: 5-app-infra/README.md [hunk 13]
```diff
@@ -1,12 +1,67 @@
...
-This Compute Engine instance will be created using the base network created during step 3-networks to access private services.
+This Compute Engine instance is created using the base network created during step `3-networks` to access private services.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:64]:
```suggestion
This Compute Engine instance is created using the base network from step `3-networks` and is used to access private services.
```

File: 5-app-infra/README.md [hunk 14]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+1. Push your master branch to the new repo.
+   ```
+   git push --set-upstream origin master
+   ```
+1. Navigate out of the repo.
+   ```
+   cd ..
+   ```
 1. Clone repo `gcloud source repos clone bu1-example-app --project=prj-bu1-c-infra-pipeline-<random>`. (this is from the terraform output from the previous section, run `terraform output cloudbuild_project_id` in the `4-projects/business_unit_1/shared` folder)
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:108]:
```suggestion
1. Clone repo `gcloud source repos clone bu1-example-app --project=prj-bu1-c-infra-pipeline-<random>`. (this is from the Terraform output from the previous section, run `terraform output cloudbuild_project_id` in the `4-projects/business_unit_1/shared` folder)
```

File: 5-app-infra/README.md [hunk 15]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+1. Change freshly cloned repo and change to non-master branch.
+   ```
+   git checkout -b plan
+   ```
+1. Copy contents of foundation to new repo.
+   ```
+   cp -RT ../terraform-example-foundation/5-app-infra/ .
+   ```
+1. Copy cloud build configuration files for terraform.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:121]:
```suggestion
1. Copy Cloud Build configuration files for Terraform.
```

File: 5-app-infra/README.md [hunk 16]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+   cp ../terraform-example-foundation/build/cloudbuild-tf-* .
+   ```
+1. Copy terraform wrapper script to the root of your new repository.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:125]:
```suggestion
1. Copy Terraform wrapper script to the root of your new repository.
```

File: 5-app-infra/README.md (hunk 1 above)
Comment (chris-ingrisani) [inline 5-app-infra/README.md:146]:
```suggestion
 1. Review the plan output in your Cloud Build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
```

File: 5-app-infra/README.md (hunk 2 above)
Comment (chris-ingrisani) [inline 5-app-infra/README.md:152]:
```suggestion
1. Review the apply output in your Cloud Build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md (hunk 3 above)
Comment (chris-ingrisani) [inline 5-app-infra/README.md:158]:
```suggestion
1. Review the apply output in your Cloud Build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md (hunk 4 above)
Comment (chris-ingrisani) [inline 5-app-infra/README.md:164]:
```suggestion
1. Review the apply output in your Cloud Build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_INFRA_PIPELINE_PROJECT
```

File: 5-app-infra/README.md [hunk 17]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
+1. Merge changes to production branch
+   ```
+   git checkout -b production
+   git push origin production
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
 
 ### Run terraform locally
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:166]:
```suggestion
### Run Terraform locally
```

File: 5-app-infra/README.md [hunk 18]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
 
 1. Change into 5-app-infra folder.
 1. Run `cp ../build/tf-wrapper.sh .`
 1. Run `chmod 755 ./tf-wrapper.sh`.
 1. Rename `common.auto.example.tfvars` to `common.auto.tfvars` and update the file with values from your environment and bootstrap.
-1. Update backend.tf with your bucket from infra pipeline example. You can run
-```for i in `find -name 'backend.tf'`; do sed -i 's/UPDATE_ME/<YOUR-BUCKET-NAME>/' $i; done```.
+1. Update backend.tf with your bucket from infra pipeline example.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:172]:
```suggestion
1. Update `backend.tf` with your bucket from the infra pipeline example.
```

File: 5-app-infra/README.md [hunk 19]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+   ```
+   for i in `find -name 'backend.tf'`; do sed -i 's/UPDATE_ME/<YOUR-BUCKET-NAME>/' $i; done
+   ```
 
 We will now deploy each of our environments(development/production/non-production) using this script.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:177]:
```suggestion
We will now deploy each of our environments (development/production/non-production) using this script.
```

File: 5-app-infra/README.md [hunk 20]
```diff
@@ -18,46 +73,109 @@ This Compute Engine instance will be created using the base network created duri
...
+1. Merge changes to production branch
+   ```
+   git checkout -b production
+   git push origin production
+   ```
+1. Review the apply output in your cloud build project https://console.cloud.google.com/cloud-build/builds?project=YOUR_CLOUD_BUILD_PROJECT_ID
 
 ### Run terraform locally
 
 1. Change into 5-app-infra folder.
```
Comment (chris-ingrisani) [inline 5-app-infra/README.md:168]:
```suggestion
1. Change into the  `5-app-infra` folder.
```

**Comment by chris-ingrisani** [comment]
@bharathkkb review complete

**Comment by bharathkkb** [comment]
Thanks for the fixes @daniel-cit @chris-ingrisani

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**